### ISSUE TITLE
unbreakable and multi_cells

### ISSUE THREAD

**Issue by yohanboniface** [PyFPDF/fpdf2#111]
I'm having weird page breaks, and it seems it's related to multi_cell + unbreakable use.

Here is a test case, based on the tutorial:

```python
import logging

from fpdf import fpdf

fpdf.LOGGER.setLevel(logging.DEBUG)
fpdf.LOGGER.addHandler(logging.StreamHandler())

data = (
    ("First name", "Last name", "Age", "City"),
    ("Jules", "Smith", "34", "San Juan"),
    ("Mary", "Ramos", "45", "Orlando"),
    ("Carlson", "Banks", "19", "Los Angeles"),
    ("Lucas", "Cimon", "31", "Saint-Mahturin-sur-Loire"),
)

pdf = fpdf.FPDF()
pdf.add_page()
pdf.set_font("Times", size=16)
line_height = pdf.font_size * 2
col_width = pdf.epw / 4  # distribute content evenly
for i in range(5):  # repeat table 4 times
    with pdf.unbreakable() as pdf:
        fpdf.LOGGER.debug(f"unbreakable block {i+1}")
        for row in data:  # data comes from snippets on the Tables documentation page
            for datum in row:
                pdf.multi_cell(col_width, line_height, f"{datum} ({i})", border=1, ln=3)
            pdf.ln(line_height)
    pdf.ln(line_height * 2)
pdf.output("unbreakable_tables.pdf")
```

This outputs this pdf:

[unbreakable_tables.pdf](https://github.com/PyFPDF/fpdf2/files/6035763/unbreakable_tables.pdf)

The first four table are OK, but there is an unexpected page break after the 4th table.

Here is the logger output:

```
unbreakable block 1
unbreakable block 2
unbreakable block 3
unbreakable block 4
Page break on page 1 at y=269 for element of height 11
Page break on page 2 at y=269 for element of height 11
Page break on page 3 at y=269 for element of height 11
Page break on page 4 at y=269 for element of height 11
Page break on page 5 at y=280 for element of height 11
Page break on page 6 at y=280 for element of height 11
Page break on page 7 at y=280 for element of height 11
Page break on page 8 at y=280 for element of height 11
Page break on page 9 at y=292 for element of height 11
Page break on page 10 at y=292 for element of height 11
Page break on page 11 at y=292 for element of height 11
Page break on page 12 at y=292 for element of height 11
Performing page jump due to unbreakable height
Page break on page 1 at y=247 for element of height 3260
Page break on page 2 at y=66 for element of height 3260
unbreakable block 5
Final doc sections size summary:
- header.size: 9.0B
- pages.size: 3.3KiB
- resources.fonts.size: 98.0B
- resources.images.size: 0.0B
- resources.dict.size: 104.0B
- info.size: 55.0B
- catalog.size: 104.0B
- xref.size: 250.0B
- trailer.size: 64.0B
```

I cannot easily understand this log, but I find quite weird that we see 12 pages in a row for a 3 rows file.

Thanks for any input on this! :)

**Comment by Lucas-C** [comment]
Thank you for the minimal bug-reproducing code!
This always makes things easier for maintainers 😊

I'll try to have a look at this today

**Comment by Lucas-C** [comment]
I found the fix and opened https://github.com/PyFPDF/fpdf2/pull/112 to solve this issue

It will be included in the next release that will occur before the end of the week

**Comment by yohanboniface** [comment]
Thanks! :)

**Comment by Lucas-C** [comment]
Released as part of `v2.3.1`: https://github.com/PyFPDF/fpdf2/releases/tag/2.3.1
